src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] table;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+    static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public int hash(K key) {
+        int result = 17;
+        result += 31 * (key == null ? 0 : key.hashCode());
+        return result;
+    }
+
+    private void increaseCapacity(int oldCapacity) {
+        int newCapacity = oldCapacity * 2;
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCapacity];
+        for (Node<K, V> oldNode : table) {
+            while (oldNode != null) {
+                Node<K, V> next = oldNode.next;
+                int newIndex = Math.abs(oldNode.hash) % newCapacity;
+                oldNode.next = newTable[newIndex];
+                newTable[newIndex] = oldNode;
+                oldNode = next;
+            }
+        }
+        table = newTable;
+        threshold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = Math.abs(hash) % table.length;
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            newNode.next = table[index];
+            table[index] = newNode;
+        }
+        size++;
+        if (size > threshold) {
+            increaseCapacity(table.length);
+        }
```

Comment:
```suggestion
        if (size == threshold) {
            increaseCapacity(table.length);
        }
```
This condition should be at the beggining as we should resize our array first if needed

Reply:
agree